models/intermediate/int_transform_contract_code.sql

```diff
@@ -0,0 +1,175 @@
+{{
+    config(
+        materialized = 'table',
+        unique_key = ['ledger_key_hash', 'closed_at'],
+        cluster_by = ["ledger_key_hash", "closed_at", "row_hash"],
+        tags = ["soroban_analytics", "intermediate", "daily"]
+    )
+}}
+
+/*
+Model: int_transform_contract_code
+
+Description:
+------------
+This intermediate model combines incremental loading and transformations for contract code data.
+It processes new or updated data, calculates new fields, and performs deduplication.
+
+Load Type: Truncate and Reload
+
+Features:
+---------
+1. Uses execution_date from Airflow to determine the processing window.
+2. Fetches data till the previous day from the Staging layer, one key hash per day (latest).
+3. Calculates `contract_create_ts` and `contract_delete_ts` from staging data.
+4. Joins the calculated timestamps with the incremental data and applies transformations.
+5. Calculates a row hash using SHA256 on all relevant fields.
+6. Performs deduplication based on the row hash of the fully transformed data. This step primarily detects duplicates in consecutive rows.
+
+Usage:
+------
+Compile:
+    dbt compile --models int_transform_contract_code
+
+Run:
+    dbt run --models int_transform_contract_code --full-refresh
+    dbt run --models int_transform_contract_code
+    dbt run --models int_transform_contract_code --vars '{"execution_date": "2024-10-01T00:00:00+00:00"}'
+*/
+
+-- Set the execution date (use Airflow value, fallback to dbt default if absent)
+{% set execution_date = var('execution_date', dbt_airflow_macros.ts()) %}
+
+with
+    source_data as (
+        select
+            contract_code_hash
+            , ledger_key_hash
+            , last_modified_ledger
+            , ledger_entry_change
+            , ledger_sequence
+            , deleted
+            , closed_at
+            , n_instructions
+            , n_functions
+            , n_globals
+            , n_table_entries
+            , n_types
+            , n_data_segments
+            , n_elem_segments
+            , n_imports
+            , n_exports
+            , n_data_segment_bytes
+            , batch_id
+            , batch_run_date
+            , cast('{{ execution_date }}' as timestamp) as airflow_start_ts
+            -- Rank records to identify the latest entry
+            , row_number() over (
+                partition by ledger_key_hash, cast(closed_at as date)
+                order by closed_at desc, ledger_sequence desc
```

Comment:
nit: I don't think you need both `closed_at` and `ledger_sequence`. They should both provide the same ordering

Reply:
Yeah. I didn't notice any such data but kept ledger sequence as a fail safe in case anything weird happens.